database.go

```diff
@@ -1138,11 +1143,12 @@ func (rp *RetentionPolicy) UnmarshalJSON(data []byte) error {
 
 // retentionPolicyJSON represents an intermediate struct for JSON marshaling.
 type retentionPolicyJSON struct {
-	Name        string        `json:"name"`
-	ReplicaN    uint32        `json:"replicaN,omitempty"`
-	SplitN      uint32        `json:"splitN,omitempty"`
-	Duration    time.Duration `json:"duration,omitempty"`
-	ShardGroups []*ShardGroup `json:"shardGroups,omitempty"`
+	Name               string        `json:"name"`
+	ReplicaN           uint32        `json:"replicaN,omitempty"`
+	SplitN             uint32        `json:"splitN,omitempty"`
```

Comment:
`SplitN` should no longer be a setting on retention policies. They can't set this themselves as it's determined by the number of data nodes in a cluster and their RF.